python/src/iceberg/expressions/base.py

```diff
@@ -122,3 +124,37 @@ def __le__(self, other):
 
     def __ge__(self, other):
         return self.value >= other.value
+
+
+class Accessor(Generic[T], ABC):
+    """An accessor for a specific position in a container of type T
+
+    Args:
+        Generic[T]: The type of container that can be accessed using this Accessor (The container must implement the StructProtocol)
+    """
+
+    def __init__(self, position: int, iceberg_type: IcebergType):
+        self._position = position
+        self._iceberg_type = iceberg_type
+
+    @property
+    def position(self):
+        """The position in the container to access"""
+        return self._position
+
+    @property
+    def iceberg_type(self):
+        """The IcebergType of the value at the accessor position"""
+        return self._value_type
+
+    @abstractmethod
+    def get(self, container: T) -> Any:
+        """Returns the value at self.position in a container of type T
+
+        Args:
+            container(T): A container to access at position `self.position`
+
+        Returns:
+            Any: The value at position `self.position` in the container
+        """
+        ...
```

Comment:
If this used the protocol, I think you could just implement it here by calling `container.get(self.position)`

Reply:
Good point, I think this is also related to my [misunderstanding](https://github.com/apache/iceberg/pull/4609#discussion_r855654244) of the abstraction

Reply:
Fixed!